jsonfeed-wp.php

```diff
@@ -11,41 +11,34 @@
 
 **************************************************************************/
 
-add_action('init', 'setup_feed_rewrite');
-add_filter('feed_content_type', 'json_feed_content_type', 10, 2);
-
-// I don't know if there is a cleaner way to add this. There is a hook 
-// feed_links_show_posts_feed, but that only controls where the main RSS 
-// feed is shown or not.
-add_action('wp_head', 'json_feed_link');
-
-function setup_feed_rewrite()
-{
-    // Register our function as the feed generator for the /feed/json URL
-    add_feed('json', 'generateJSONFeed');
+// Flush the rewrite rules to enable the json feed permalink
+register_activation_hook( __FILE__, 'json_feed_setup_rewrite' );
+function json_feed_setup_rewrite() {
+	json_feed_setup_feed();
```

Comment:
Be sure to call `add_feed()` which adds the [required rewrite](https://github.com/WordPress/WordPress/blob/dd6da701b286579819cd6aa518aa2d7018efd759/wp-includes/rewrite.php#L248) rule before re-generating the saving the rewrite rules.